docs/proposals/03-loadbalancer.md

```diff
@@ -0,0 +1,128 @@
+
+## Summary
+Network service for Load Balancing is needed in every modern infrastructure (e.g. Kubernetes). Load balancing makes services scalable, fault-tolerant and accessible from external users. The load balancer described here is a L3-Loadbalancer (so it implements an IP level load balancing). The load balancer function is a standard network function that can be to be deployed (as an implementation detail) on a platform/node that is capable of running network functions (e.g. dp-service enabled nodes/nic's) 
+
+## Motivation
+The onMetal network is a fully routed network. To run workloads hosted on several nodes (in Kubernetes terms equivalent to cluster IP's) and not only on a single node (node IP's) solutions like ECMP (Equal cost multi pathing) seem to be the natural solution. But ECMP has certain limitations: the amount of nodes it can be used to distribute traffic to and a hash based algorithm to pin the network flows, that changes - based on the amount of targets - its distributed pinning. 
+To avoid those limitations and to have better control over the load balancer behavior an onMetal load balancer service needs to be provided.
+
+### Goals
+- define a user facing API (onMetal api) for the load balancer
+- enable the user to request an L3 Loadbalancer (only IP addresses) with load balancer address (outside facing) and 
+- with multiple target's based on the onMetal ([OEP-1](01-networking-integration.md)) network interface architecture
+- the targets of the load balancer must be dynamic changeable
+- the load balancer needs to avoid forwarding unintended traffic.
+  - based on ports/protocols (UDP/TCP/SCTP) it must filter what needs to be load balanced
+  - ICMP requests are not part of forwarded traffic and will be filtered by the load balancer anyhow (the loa)
+- Load balancer (IP) and targets can be parts of different networks but
+- all targets (NetworkInterfaces) must be of the same network 
+
+### Non-Goals
+- no address or port translation / rewriting (no SNAT / DNAT) (L4 Loadbalancer) support
+- no injection of additional information (e.g. x-forwarded-for) (L7 Loadbalancer) support
+- no protocol offloading like ssl (L7 Loadbalancer) support
+- if multiple load balancer IP's are needed, request multiple load balancers (a load balancer services one load balancer IP)
+
+### Details
+- load balancing is used to deliver a packet addressed to the load balancer to one of its targets via the onMetal network routing
+- the load balancing will be for the target and the source transparent
+- the target needs to know the load balancers IP and needs to answer with it (and to receive traffic with it)
+- answers to request will be directly delivered since all details are known to the target
+- payload packages are not changed, to not lose any information
+
+## Proposal
+A network loadbalancer CRD allows the user to define the network function. all entries except the `targetNetworkInterface` must be immutable. 
+
+### User defined object
```

Comment:
In most clouds it is possible to define and request a load balancing objects as `Service` resources with certain types like `LoadBalancer` or `NodePort` (in case of MetalLB), and with `Ingress` resource.

Probably it is worth to mention how these objects will correlate as it would allow to avoid any further compatibility and implementation issues.

Reply:
Kubernetes `Service`s of type `NodePort` are a feature of the `Node`s running in a Cluster.
`LoadBalancer`s are inherently cloud provider specific.

Reply:
I mentioned explicitly the usage of a Kubernetes ingess. hope that helpes.